packages/web/pages/docs/rest-api.mdx

```diff
@@ -47,3 +47,99 @@ You then pass your API key in each request with one of these methods:
 
 - Include the API key in the request headers: `X-API-KEY=...`
 - Include the API key as a query string parameter: `?apiKey=...`
+
+## API endpoints
+
+All API endpoints expect the body to be a JSON string and the header Content-Type to be "application/json".
+
+### /api/sendMail
+
+Send an email using your configured sendmail.
+
+**Authentication required**
+
+Yes, using an [API key](#api-keys)
+
+**Request parameters**
+
+You can either use /api/sendMail to send raw html or a template.
+
+- [Standard nodemailer options](#standard-nodemailer-options)
+- templateName: string (required if using template mode, must be the name of one of your templates)
+- props: []string (required if using template mode, must correspond to the props your template expects)
+- html: string (required if using raw html mode)
+- previewName: string (optional) - used for analytics only, gets put on the Message object in the database so you can tell what preview was used to send the email
+
+Example request (template mode):
+
+```
+POST /api/sendMail
+Content-Type: application/json
+X-API-KEY: your-api-key
+
+{
+  "to": "to@example.com",
+  "from": "you@example.com",
+  "subject": "Hello world",
+  "templateName": "myTemplate",
+  "props": {
+    "name": "Alex"
+  },
+}
+```
```

Comment:
Yeah, that would be cool

Reply:
@psugihara I played with this a bit in another branch called `382-doc-api-sendmail-code` I think that's how it could work:
<img width="1757" alt="Screenshot 2022-12-01 at 8 37 31 PM" src="https://user-images.githubusercontent.com/24866/205215960-5652c475-be00-47e6-8b7d-d427e77e7636.png">

Do you know how to get the rehype/highlight.js syntax highlighting working?  I see that it works if you use ```js in the markdown, but prettier then squashes the formatting of the code, so I need to apply it manually.  But we could easily do that in CodeBlock (in that branch) where we render the text.  I think I almost got it working, but confused about what is necessary for it -- I see that we're using rehype-highlight, but what do we need to use that outside of the context of markdown?  Probably some sub-dependency of it.

Reply:
I figured out a way to do it without doing prettier ignore and keeping the code the same way otherwise (in the mdx etc)